python/ray/train/batch_predictor.py

```diff
@@ -177,6 +192,13 @@ def __call__(self, batch):
         ray_remote_args["num_cpus"] = num_cpus_per_worker
         ray_remote_args["num_gpus"] = num_gpus_per_worker
 
+        if separate_gpu_stage and num_gpus_per_worker > 0:
+            preprocessor = self.get_preprocessor()
+            if preprocessor:
+                override_prep = BatchMapper(lambda x: x)
```

Comment:
doesnt seem like override_prep is being used anywhere?

Reply:
Oh, that's weird LINT didn't catch it.

Reply:
Oh it's actually a local variable from the previous PR.

Reply:
I think there might be a `set_preprocessor` line missing to disable the preprocessor for the predictor.

Reply:
It's actually correct--- I modified the unit test. If you remove this line, the test will fail. Also added a clarifying comment.